src/store/index.js

```diff
@@ -7,7 +7,9 @@ export function observe(key, id, f) {
   subscribe(`${key}-${id}`, (_, oldState, newState) => {
     const m = getEntity(newState, 'lock', id);
     const oldM = getEntity(oldState, 'lock', id);
-    if (m != oldM) f(m);
+    if (m && !m.equals(oldM)) {
```

Comment:
Perhaps (m != null) would be safer?  What types does getEntity return?

Reply:
they will always be different because `m` is an immutable object (which is a new reference, so will always be false if you compare with !=). that's why the fix is actually use `.equals`, which is an immutable method.